### ISSUE TITLE
select

### ISSUE THREAD

**Issue by felix-bohlin** [felix-bohlin/ui#29]
- [x] expand accessibility section with more info
- [x] grouped selects should use role="group"
- [x] validation

**Comment by hermit99** [comment]
Looks like Select is still work in progress, otherwise broken? Here are some warnings that may help adjust the design:

![image](https://github.com/user-attachments/assets/9f19b0bd-dbea-4d33-89d3-34b7ddd45322)

**Comment by felix-bohlin** [comment]
It's actually valid syntax albeit very [experimental](https://developer.chrome.com/blog/rfc-customizable-select)!

I haven't really figured out how to make Vite understand that it is valid though. Since it's just a warning I haven't given it much effort.

**Comment by hermit99** [comment]
Interesting read, thanks @felix-bohlin. Warnings aside, my point was that Select component seems broken?
![image](https://github.com/user-attachments/assets/95e61b3f-c15c-4f15-b240-f59bb17d0873)

**Comment by felix-bohlin** [comment]
Hmmm, I'm not getting that. What OS, browser and browser version are you running? 🤔 

https://github.com/user-attachments/assets/44c1057a-3b5c-4265-a478-4763c26684b3

**Comment by hermit99** [comment]
Win 11.  Consistent on Chromium and Firefox

**Comment by felix-bohlin** [comment]
Ah, looks like you need to go to chrome://flags and enable `Experimental Web Platform features`.

**Comment by hermit99** [comment]
> Ah, looks like you need to go to chrome://flags and enable `Experimental Web Platform features`. I turned it off for me and it looked identical to your screenshot.

A bit too modern then 😄

**Comment by felix-bohlin** [comment]
Haha yeah. While the purpose with this repo is to explore the possibilities of a next-gen UI library, moments like this make me wonder sometimes if there should be one version that works without feature flags, and another that is next-gen. The way to just copy and paste components is nice enough to justify that, but I know it would be too much work to maintain 😅

**Comment by hermit99** [comment]
It would be nice to have a basic working version in parallel with the Stage 2 proposal, which may take years to become standard, but I know it's easier said than done.

**Comment by felix-bohlin** [comment]
I hear you @hermit99 🙂 I'll look into it, it shouldn't be too hard I think. The only difference should be the option list which, in that case, would use the browser standard one.

**Comment by felix-bohlin** [comment]
There @hermit99, I've added support for selects with native option lists now! [Here's the commit ](https://github.com/felix-bohlin/ui/commit/0775d27ddce6a6b3196524ba8232d055a6bf416e)with the included changes. Let me know if I've missed anything! :)

**Comment by hermit99** [comment]
Awesome! Thanks @felix-bohlin. Yeah, the native chevron is hard to style.

**Comment by felix-bohlin** [comment]
No problem! Although I want it to be a component library of the future, having some simple progressive enhancement built-in probably won't hurt. I guess I'll know when I get there 😅

**Comment by hermit99** [comment]
> There @hermit99, I've added support for selects with native option lists now! [Here's the commit ](https://github.com/felix-bohlin/ui/commit/0775d27ddce6a6b3196524ba8232d055a6bf416e)with the included changes. Let me know if I've missed anything! :)
> 
> EDIT: I noticed just now that the chevron isn't the same 🙃

If we want the chevron closer to the native one, we could do something like this?
```diff 
select.css line 142-154
- block-size: 0;
- border-block-start: 5px solid;
- border-inline: 5px solid transparent;
- content: "";
- display: inline-block;
- flex-shrink: 0;
- grid-column: 1/-1;
- grid-row: 1;
- inline-size: 0;
- inset-inline-end: var(--size-3);
- justify-self: end;
- pointer-events: none;
- position: relative;

+ border-style: solid;
+ border-width: 0.2em 0.2em 0 0;
+ content: '';
+ display: inline-block;
+ grid-column: 1/-1;
+ grid-row: 1;
+ height: 0.45em;
+ inset-inline-end: var(--size-3);
+ justify-self: end;
+ pointer-events: none;
+ position: relative;
+ transform: rotate(135deg);
+ width: 0.45em;
```
Just some thoughts.

**Comment by felix-bohlin** [comment]
> If we want the chevron closer to the native one, we could do something like this?

Thanks, although I'm currently going for the [Material Design 3 look](https://m3.material.io/components/text-fields/guidelines#7719e269-eeea-4be8-8c03-510e0b52ba75) for the chevrons, so I fixed it. 🤓
